demo-angular/app/pages/login/login.component.ts

```diff
@@ -34,9 +56,61 @@ export class LoginComponent {
         });
     }
 
-    public getCurrentAccessToken() {
+    getCurrentAccessToken() {
         let accessToken = Facebook.getCurrentAccessToken();
 
         alert("Current access token: " + JSON.stringify(accessToken, null, '\t'));
     }
+
+    generateLinksShareContent() {
+        return Facebook.createLinksShareContent('https://www.nativescript.org',
+            'Create Native iOS and Android Apps With JavaScript',
+            {
+                hashtag: '#Nativescript'
+            })
+    }
+
+    generatePhotosShareContent() {
+        const logoImage = fromResource('logo');
+        return Facebook.createPhotosShareContent([logoImage], false, {
+            hashtag: '#Nativescript'
+        });
+    }
+
+    generateGenericTemplateContent() {
+        return Facebook.createShareMessengerGenericTemplateContent({
+            element: {
+                title: 'Nativescript',
+                subtitle: 'Create Native iOS and Android Apps With JavaScript',
+                imageUrl: 'https://d2odgkulk9w7if.cloudfront.net/images/default-source/home/how-it-works-min.png',
+                button: {
+                    title: 'Check Doc',
+                    url: 'https://docs.nativescript.org'
+                },
+                defaultAction: {
+                    title: 'Go HomePage',
+                    url: 'https://www.nativescript.org'
+                }
+            },
+            // it seems android have to provide a pageId, otherwise the MessageDialog just wont show
+            pageID: 'testestsett',
+            imageAspectRatio: Facebook.MessageGenericTemplateImageAspectRatio.Horizontal
```

Comment:
When running the demo app, I get an error that `MessageGenericTemplateImageAspectRatio` is not recognized. Can you add the enums `MessageGenericTemplateImageAspectRatio` and `MessageMediaTemplateMediaType` to share-manager.d.ts?

Reply:
I've added those enums to the d file. I don't know if it will work because I can't reproduce this issue.